oxd-server/src/test/java/org/gluu/oxd/server/CheckAccessTokenTest.java

```diff
@@ -25,6 +26,7 @@ public void test(String host, String redirectUrls, String userId, String userSec
         ClientInterface client = Tester.newClient(host);
         String nonce = CoreUtils.secureRandomString();
         RegisterSiteResponse site = RegisterSiteTest.registerSite(client, opHost, redirectUrls);
+        SeleniumTester.enableTrustedClientOnGluuServer(opHost, userId, userSecret, site.getClientId(), site.getClientSecret());
```

Comment:
It's kind of workaround because we go to oxtrust and enable trusted flag via admin UI. But here we don't want to enable that flag. We want to login without it. It means that we should remove this line and instead inside `GetTokensByCodeTest.tokenByCode` method add navigation to `Authorization Endpoint` of OP directly (without UI) and emulate user login (you have it already done in SeleniumTester but instead need to hit `Authorization Endpoint` directly, without oxtrust).

```
authorizationEndpoint += "?response_type=code&state=bla&nonce=bla&client_id=" + site.getClientId();
        authorizationEndpoint += "&redirect_uri=" + redirectUri;
        authorizationEndpoint += "&scope=" + Utils.joinAndUrlEncode(scope);
```

Reply:
We have followed above steps. But now the `GetTokensByCodeTest.tokenByCode` test case is failing again even after successful login and allowing permission. I think as selenium and testng works separately so successful login using selenium donot have any impact on tesng test cases.

Reply:
Not sure I'm following ... Selenium driver should click login and then accept. As result within SAME selenium session we should get redirect with code. Then we should obtain token by that code.
Please refer to this oxauth client example as sample:
https://github.com/GluuFederation/oxAuth/blob/1991acd03c1733e704b0263f24f32d1e75dc2c69/Client/src/test/java/org/xdi/oxauth/BaseTest.java#L553